src/dataframe/dataframe.jl

```diff
@@ -881,10 +913,10 @@ hcat!(df1::DataFrame, df2::DataFrame;
            makeunique=makeunique, copycols=copycols)::DataFrame
 
 hcat!(df::DataFrame, x::AbstractVector; makeunique::Bool=false, copycols::Bool=true) =
-    hcat!(df, DataFrame(AbstractVector[x], copycols=copycols),
+    hcat!(df, DataFrame(:x1 => x, copycols=copycols),
```

Comment:
Wouldn't it be more direct to pass `AbstractVector[x], [:x1]`?

Reply:
ok - fixed